From 42d1f377e296c6cb2c46d4c9d1e669d7448998a0 Mon Sep 17 00:00:00 2001
From: zhangyunlong <dalon.zhang@rock-chips.com>
Date: Mon, 14 Aug 2017 17:14:48 +0800
Subject: [PATCH] camera: rockchip: camsys_drv v0.0x22.3

switch TX1/RX1 D-PHY of rk3288/3399 to RX status before it's
initialization to avoid conflicting with sensor output.

Change-Id: I672730fe5fb5a33b8437df1ae61078a9a79ac41b
Signed-off-by: zhangyunlong <dalon.zhang@rock-chips.com>
---
 drivers/media/video/rk_camsys/camsys_internal.h   | 5 ++++-
 drivers/media/video/rk_camsys/camsys_soc_rk3288.c | 2 ++
 drivers/media/video/rk_camsys/camsys_soc_rk3399.c | 3 ++-
 3 files changed, 8 insertions(+), 2 deletions(-)

diff --git a/drivers/media/video/rk_camsys/camsys_internal.h b/drivers/media/video/rk_camsys/camsys_internal.h
index 0c533d223117..86d571ebb13c 100644
--- a/drivers/media/video/rk_camsys/camsys_internal.h
+++ b/drivers/media/video/rk_camsys/camsys_internal.h
@@ -172,8 +172,11 @@
 	1) modify the condition of DRM iommu, which makes code  more readable
 	by using of_parse_phandle to check whether the "iommus" phandle exists
 	in the isp device node.
+*v0.0x22.3:
+	1) switch TX1/RX1 D-PHY of rk3288/3399 to RX status before
+	it's initialization to avoid conflicting with sensor output.
 */
-#define CAMSYS_DRIVER_VERSION                   KERNEL_VERSION(0, 0x22, 2)
+#define CAMSYS_DRIVER_VERSION                   KERNEL_VERSION(0, 0x22, 3)
 
 #define CAMSYS_PLATFORM_DRV_NAME                "RockChip-CamSys"
 #define CAMSYS_PLATFORM_MARVIN_NAME             "Platform_MarvinDev"
diff --git a/drivers/media/video/rk_camsys/camsys_soc_rk3288.c b/drivers/media/video/rk_camsys/camsys_soc_rk3288.c
index e8ad148e9248..91331a60b52e 100755
--- a/drivers/media/video/rk_camsys/camsys_soc_rk3288.c
+++ b/drivers/media/video/rk_camsys/camsys_soc_rk3288.c
@@ -345,6 +345,8 @@ camsys_dev_t *camsys_dev, camsys_soc_cfg_t cfg_cmd, void *cfg_para)
 		para_int = (unsigned int *)cfg_para;
 		__raw_writel((((*para_int) & 0x03) << 3)|(0x03 << 3),
 			(void *)(camsys_dev->rk_grf_base + 0x01d4));
+		/* TX1/RX1 DPHY switch to RX status */
+		__raw_writel(0xa000a000, (void *)(camsys_dev->rk_grf_base + 0x027c));
 		break;
 	}
 
diff --git a/drivers/media/video/rk_camsys/camsys_soc_rk3399.c b/drivers/media/video/rk_camsys/camsys_soc_rk3399.c
index b7d001c48e9d..749d2953913f 100644
--- a/drivers/media/video/rk_camsys/camsys_soc_rk3399.c
+++ b/drivers/media/video/rk_camsys/camsys_soc_rk3399.c
@@ -368,7 +368,8 @@ int camsys_rk3399_cfg
 		para_int = (unsigned int *)cfg_para;
 		__raw_writel((((*para_int) & 0x03) << 3) | (0x03 << 3),
 		(void *)(camsys_dev->rk_grf_base + 0x204));
-		/* set 0xffffffff to max all */
+		/* TX1/RX1 DPHY switch to RX status */
+		__raw_writel(0x300020, (void *)(camsys_dev->rk_grf_base + 0x6260));
 		break;
 	}
 
-- 
2.35.3

